### ISSUE TITLE
Deleting Orphaned Objects during Realm Migration

### ISSUE THREAD

**Issue by saltyskip** [realm/realm-cocoa#7139]
## Goals

I want to migrate existing data base stores to embedded. There maybe some objects that have been unintentionally orphaned due to developer error. I want to migrate the objects to embedded and either ignore or delete orphaned objects in the process

## Expected Results

Successful migration, if there are any objects that do not have a parent object they should be deleted.

## Actual Results

```
Fatal error: Realm has crashed At least one object does not have a backlink (data would get lost).: file Bitcoin_com/AppDelegate.swift, line 98
```

## Steps for others to Reproduce

Create several parent child relationships. Create an orphaned child object

Parent A -> Child B
Child C

When you run the migration you will get the error above due to the existance of child C. In migration block I want to delete child C if that is the case.

## Version of Realm and Tooling

Realm framework version: Realm 10.7.0

Xcode version: 12.4

iOS/OSX version: 14.4

**Comment by DominicFrei** [comment]
@saltyskip This change came with https://github.com/realm/realm-core/pull/4414 in core. Please also refer to the discussion (and all the linked issues) in there.

TL;DR
There was a bug where users could not migrate from `Object` to `EmbeddedObject` due to missing or multiple backlinks.
One of the decisions that had to be made during that change was the handling of orphaned objects. There were two possible solutions:
(1) Silently delete objects that would now be orphaned after the switch to `EmbeddedObject` without the developer noticing that data gets lost.
(2) Throw an error to warn the developer about orphaned objects that would otherwise be lost after that switch.

We chose to go the safer path of option (2).

When migrating to an `EmbeddedObject` you have to make sure that at the end of the migration (which is when the actual change happens) every object has one and only one backlink.

**Comment by saltyskip** [comment]
Hi I see, I've updated my codes to manually remove the orphaned objects, it might be nice to either have a configuration setting to delete orphans automatically or update documentation examples of migrations to show how to safely delete orphaned objects during migration.

I think its safe to delete this issue. Thanks @DominicFrei

**Comment by DominicFrei** [comment]
Thanks for letting us know, @saltyskip.

I did create a ticket to include a helper function to make deleting those orphaned objects easier: #7145 

I've also noted that an update of the documentation would be a good idea. 👍
